src/werkzeug/http.py

```diff
@@ -1298,6 +1303,9 @@ def dump_cookie(
         if samesite not in {"Strict", "Lax", "None"}:
             raise ValueError("SameSite must be 'Strict', 'Lax', or 'None'.")
 
+    if partitioned and not secure:
```

Comment:
Would it make sense to automatically force `secure=True` for `partitioned=True`?

Reply:
I'm not sure, would it be confusing if `dump_cookie(..., partitioned=True, secure=False)` worked even though it shouldn't?

Reply:
We could make `secure=None` by default to detect if `False` is passed explicitly, but I don't really think it's worth it. If the docs for the `partitioned` argument say "This forces `secure=True`." I think that's enough.

Reply:
Ok, I've changed.